services/src/main/java/org/keycloak/services/resources/admin/IdentityProvidersResource.java

```diff
@@ -166,21 +170,81 @@ public Map<String, String> importFrom(@Parameter(description = "JSON body") Map<
     }
 
     /**
-     * Get identity providers
+     * List identity providers.
      *
-     * @return
+     * @param search Filter to search specific providers by name. Search can be prefixed (name*), contains (*name*) or exact (\"name\"). Default prefixed.
+     * @param briefRepresentation Boolean which defines whether brief representations are returned (default: false)
+     * @param firstResult Pagination offset
+     * @param maxResults Maximum results size (defaults to 100)
+     * @return The list of providers.
      */
     @GET
     @Path("instances")
     @NoCache
     @Produces(MediaType.APPLICATION_JSON)
     @Tag(name = KeycloakOpenAPI.Admin.Tags.IDENTITY_PROVIDERS)
-    @Operation( summary = "Get identity providers")
-    public Stream<IdentityProviderRepresentation> getIdentityProviders() {
+    @Operation(summary = "List identity providers")
+    public Stream<IdentityProviderRepresentation> getIdentityProviders(
+            @Parameter(description = "Filter specific providers by name. Search can be prefix (name*), contains (*name*) or exact (\"name\"). Default prefixed.") @QueryParam("search") String search,
+            @Parameter(description = "Boolean which defines whether brief representations are returned (default: false)") @QueryParam("briefRepresentation") Boolean briefRepresentation,
+            @Parameter(description = "Pagination offset") @QueryParam("first") Integer firstResult,
+            @Parameter(description = "Maximum results size (defaults to 100)") @QueryParam("max") Integer maxResults) {
+        this.auth.realm().requireViewIdentityProviders();
+
+        if (maxResults == null) {
+            maxResults = 100; // always set a maximum of 100
+        }
+
+        Function<IdentityProviderModel, IdentityProviderRepresentation> toRepresentation = briefRepresentation != null && briefRepresentation
+                ? m -> ModelToRepresentation.toBriefRepresentation(realm, m)
+                : m -> StripSecretsUtils.strip(ModelToRepresentation.toRepresentation(realm, m));
+
+        Stream<IdentityProviderModel> stream = realm.getIdentityProvidersStream().sorted(new IdPComparator());
+        if (!StringUtil.isBlank(search)) {
+            stream = stream.filter(predicateByName(search));
+        }
+        if (firstResult != null) {
+            stream = stream.skip(firstResult);
+        }
+        return stream.limit(maxResults).map(toRepresentation);
+    }
+
+    /**
+     * Count identity providers.
+     *
+     * @param search Filter to search specific providers by name. Search can be prefixed (name*), contains (*name*) or exact (\"name\"). Default prefixed.
+     * @return Integer count of providers.
+     */
+    @GET
+    @Path("instances/count")
+    @NoCache
+    @Produces(MediaType.APPLICATION_JSON)
+    @Tag(name = KeycloakOpenAPI.Admin.Tags.IDENTITY_PROVIDERS)
+    @Operation( summary = "Count identity providers")
+    public Integer countIdentityProviders(
```

Comment:
Sorry I did not spot it before, but do we really need count endpoint?

I know that we have "count" for some objects, but for instance, I've checked clients (which already support pagination). And the `ClientsResource` does not have any "count" endpoint for clients. It seems that admin console doesn't need it as it does not display the count of search results and count of pages for clients. It just display arrow for next page (if exists).

IMO if admin console doesn't need "count" endpoint, then my vote would be to skip "count" endpoint. I would just go for necessary minimum to have admin console working (Especially since we may migrate to "Admin REST API v2" and hence adding more unnecessary mess to admin REST API doesn't help...)

Reply:
Ah! Okis, the I will remove the `count` endpoint. I just added it because I cloned the behavior in the users endpoints.

Reply:
Yeah, we're not using this at the moment. Perhaps we should, but pagination is just kinda borked in general atm. That won't change until we start working on a proper V2 API.

Reply:
Removed the count endpoint!